### PR TITLE
50238_profile_picture_upload_option_for_user

### PR DESCRIPTION
**Pull request by mdmahmudurrahman**

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -7,4 +7,13 @@ class Category < ActiveRecord::Base
     uniqueness: {case_sensitive: false}
 
   mount_uploader :picture, ImageUploader
+
+  validate :picture_size
+
+  private
+  def picture_size
+    if picture.size > 3.megabytes
+      errors.add(:picture, "should be less than 3MB")
```
Comment (dieunb) [inline app/models/category.rb]: i18n

File: app/models/user.rb [hunk 2]
```diff
@@ -23,4 +23,14 @@ class User < ActiveRecord::Base
...
             through: :passive_relationships,
             source: :follower
+
+  mount_uploader :picture, ImageUploader
+  validate :picture_size
+
+  private
+  def picture_size
+    if picture.size > 3.megabytes
+      errors.add(:picture, "should be less than 3MB")
```
Comment (dieunb) [inline app/models/user.rb]: i18n

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -5,4 +5,22 @@ class UsersController < ApplicationController
   def index
     @users = @users.page(params[:page]).per 10
   end
+    
+  def edit
+  end
+
+  def update
+    if @user.update(user_params)
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: app/models/user.rb [hunk 4]
```diff
@@ -23,4 +23,14 @@ class User < ActiveRecord::Base
   has_many :followers,
             through: :passive_relationships,
             source: :follower
+
+  mount_uploader :picture, ImageUploader
+  validate :picture_size
+
+  private
+  def picture_size
```
Comment (dieunb) [inline app/models/user.rb]: duplicated code, should split it to new module, and include it on this model

File: app/models/category.rb [hunk 5]
```diff
@@ -7,4 +7,13 @@ class Category < ActiveRecord::Base
     uniqueness: {case_sensitive: false}
 
   mount_uploader :picture, ImageUploader
+
+  validate :picture_size
+
+  private
+  def picture_size
```
Comment (dieunb) [inline app/models/category.rb]: duplicated code, should split it to new module, and include it on this model

File: app/models/concerns/image_validation.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+module ImageValidation
+  def picture_size
```
Comment (dieunb) [inline app/models/concerns/image_validation.rb]: private method :D

File: app/models/ability.rb [hunk 7]
```diff
@@ -7,6 +7,8 @@ def initialize user
       can :manage, :all
     else
       can :read, :all
+      can [:edit], User, id: user.id
```
Comment (dieunb) [inline app/models/ability.rb]: `can [:edit, :update], User, id: user.id`

**Comment by sujoydatta** [comment]
ok

File: config/locales/en.yml [hunk 8]
```diff
@@ -103,3 +103,10 @@ en:
   confirm_message_delete: "Are you sure to delete"
   delete_success_message: "User deleted"
   user_removal_failure: "User not deleted"
+  update_success_message: User updated successfully
+  user_removal_failure: User not deleted
+  user_update_notice: User not updated
+  view_all: View all
+  view_all_request: View All Requests
+  welcome_message: Welcome to
+  image_size_notification: "Image size shoule be less then 3MB"
```
Comment (shakil-fg) [inline config/locales/en.yml]: make consistent locale

Add double quote to all strings

File: config/locales/vn.yml [hunk 9]
```diff
@@ -52,3 +52,16 @@ vn:
...
+  email: "E-mail"
+  action: "hoạt động"
+  edit_user: "Chỉnh sửa"
+  delete_user: "Xóa bỏ"
+  confirm_message_delete: "Bạn có chắc chắn muốn xóa"
+  delete_success_message: "dùng đã bị xóa"
+  user_removal_failure: "Người dùng không bị xóa"
+  update_success_message: "Người sử dụng cập nhật thành công"
+  user_update_notice: "Người dùng không được cập nhật"
+  image_size_notification: "Kích thước hình ảnh nên được ít hơn 3MB"
```
Comment (shakil-fg) [inline config/locales/vn.yml]: make consistent locale

Add double quote to all strings

**Pull request merged**